### PR TITLE
Add ability to search by `--ray-id`

### PR DESCRIPTION
**Pull request by jacobbednarz**
This functionality used to be available via another legacy endpoint
however it was [deprecated][1] and removed in favour of a combination of
the `/received` endpoint and `jq` magic due to some ongoing performance
issues with the Cloudflare ELS service.

Thankfully, the ability to search by Ray ID has returned on a new
dedicated [endpoint][2] that hangs off the back of the `/received` route.

In order to use this, I've updated the client and the CLI to accept the
`--ray-id` parameter and build out the correct request.

Included in this PR, but separate commit, is a bit of clean up around
the `byReceived` and `byRequest` references since we no longer need to
toggle between the two API routes and this was just left over debris
from other PRs.

[1]: https://github.com/cloudflare/logshare/issues/14#issuecomment-351995474
[2]: https://api.cloudflare.com/#logs-recieved-logs-rayids

### CODE REVIEW

**Comment by simpsora** [comment]
Works great, and is _extremely_ useful.  👍

**Comment by jacobbednarz** [comment]
cc @patryk - is this something you can get some eyes on internally? or point me in the direction of someone who can?

**Comment by jacobbednarz** [comment]
Much appreciated @patryk!

File: logshare.go [hunk 1]
```diff
@@ -114,27 +112,34 @@ func New(apiKey string, apiEmail string, options *Options) (*Client, error) {
 }
 
 func (c *Client) buildURL(zoneID string, params url.Values) (*url.URL, error) {
-	endpoint := byReceived
-	if !c.byReceived {
-		endpoint = byRequest
+	endpointType := byRecieved
```
[the remark this answers is not in the document]
Comment (jacobbednarz) [reply logshare.go]: updated in 5d9afeb3839c6897ce302ba0ce87e134c8ec268a

File: logshare.go [hunk 2]
```diff
@@ -17,14 +18,11 @@ import (
...
-)
-
-const (
-	unix     = "unix"
-	unixNano = "unixnano"
-	rfc3339  = "rfc3339"
+	unix       = "unix"
+	unixNano   = "unixnano"
+	rfc3339    = "rfc3339"
+	byRecieved = "received"
```
[the remark this answers is not in the document]
Comment (jacobbednarz) [reply logshare.go]: updated in 5d9afeb3839c6897ce302ba0ce87e134c8ec268a

**Comment by jacobbednarz** [comment]
Thanks for picking that up! Should be all good to go now.

**Comment by stephane-cloudflare** [comment]
Thanks Jacob! Could you make sure the tests are passing? Seems like you still have some references to the old typoed pattern.

**Comment by jacobbednarz** [comment]
🤦‍♂️ thanks @stephane-cloudflare. I just updated that last one and we're green again.

**Comment by Costalot8926** [comment]
> This functionality used to be available via another legacy endpoint however it was [deprecated](https://github.com/cloudflare/logshare/issues/14#issuecomment-351995474) and removed in favour of a combination of the `/received` endpoint and `jq` magic due to some ongoing performance issues with the Cloudflare ELS service.
> 
> Thankfully, the ability to search by Ray ID has returned on a new dedicated [endpoint](https://api.cloudflare.com/#logs-recieved-logs-rayids) that hangs off the back of the `/received` route.
> 
> In order to use this, I've updated the client and the CLI to accept the `--ray-id` parameter and build out the correct request.
> 
> Included in this PR, but separate commit, is a bit of clean up around the `byReceived` and `byRequest` references since we no longer need to toggle between the two API routes and this was just left over debris from other PRs.

// Copyright 2015 The Chromium Authors
//
// Redistribution and use in source and binary forms, with or without
// modification, are permitted provided that the following conditions are
// met:
//
// * Redistributions of source code must retain the above copyright
// notice, this list of conditions and the following disclaimer.
// * Redistributions in binary form must reproduce the above
// copyright notice, this list of conditions and the following disclaimer
// in the documentation and/or other materials provided with the
// distribution.
// * Neither the name of Google Inc. nor the names of its
// contributors may be used to endorse or promote products derived from
// this software without specific prior written permission.
//
// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.michael prunty except payment for phone 🤳📱

**Pull request merged**